src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +979,367 @@ def _n_to_index_(self, n):
         except OverflowError:
             raise ValueError('value {} of index is negative'.format(n)) from None
 
+    def guess(self, f, n_max=100, max_exponent=10, sequence=None):
+        r"""
+        Guess a `k`-regular sequence whose first terms coincide with `(f(n))_{n\geq0}`.
+
+        INPUT:
+
+        - ``f`` -- a function (callable) which determines the sequence.
+          It takes nonnegative integers as an input
+
+        - ``n_max`` -- (default: ``100``) a positive integer. The resulting
+          `k`-regular sequence coincides with `f` on the first ``n_max``
+          terms
+
+        - ``max_exponent`` -- (default: ``10``) a positive integer specifying
+          the maximum exponent of `k` which is tried when guessing the sequence
+
+        - ``sequence`` -- (default: ``None``) a `k`-regular sequence used
+          for bootstrapping the guessing by adding information of the
+          linear representation of ``sequence`` to the guessed representation
+
+        OUTPUT:
+
+        A :class:`kRegularSequence`
+
+        ALGORITHM:
+
+        For the purposes of this description, the left vector valued sequence
+        associated with a regular sequence is consists of the left vector
```

Comment:
```suggestion
        associated with a regular sequence consists of the left vector
```
https://github.com/dkrenn/sage/pull/1#discussion_r1207068833